src/wp-includes/class-wp-theme-json.php

```diff
@@ -1925,6 +1925,14 @@ protected static function flatten_tree( $tree, $prefix = '', $token = '--' ) {
 	 * @return array  Returns the modified $declarations.
 	 */
 	protected static function compute_style_properties( $styles, $settings = array(), $properties = null, $theme_json = null, $selector = null, $use_root_padding = null ) {
+		$args      = func_get_args();
+		$cache_key = 'compute_style_properties_' . md5( wp_json_encode( $args ) );
+		$cache     = wp_cache_get( $cache_key, 'wp-styles' );
+
+		if ( $cache ) {
+			return $cache;
+		}
```

Comment:
Certainly, you can add the code after line 1944 to avoid creating a cache for `empty( $styles )`. Please insert the code in the appropriate location to achieve this.